platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +251,261 @@ def get_sfp(self, index):
+                (ModuleBase.REBOOT_CAUSE_HARDWARE_OTHER, 'Phy reset'),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/usbphy_rst':
+                (ModuleBase.REBOOT_CAUSE_HARDWARE_OTHER, 'USB Phy reset'),
+        }
+        for f, rd in reboot_cause_map.items():
+            if utils.read_int_from_file(f) == 1:
+                return rd
+        return ModuleBase.REBOOT_CAUSE_NON_HARDWARE, ''
+
+    def get_midplane_ip(self):
+        """
+        Retrieves the midplane IP-address of the module in a modular chassis
+        When called from the Supervisor, the module could represent the
+        line-card and return the midplane IP-address of the line-card.
+        When called from the line-card, the module will represent the
+        Supervisor and return its midplane IP-address.
+        When called from the DPU, returns the midplane IP-address of the dpu-card.
+        When called from the Switch returns the midplane IP-address of Switch.
+        Returns:
+            A string, the IP-address of the module reachable over the midplane
+        """
+        if not self.midplane_ip:
+            midplane_data = DeviceDataManager.get_platform_midplane_network()
+            network_cidr = midplane_data['bridge_address']
+            ip_network_cidr = ip_network(network_cidr, strict=False)
+            self.midplane_ip = str(ip_network_cidr[self.dpu_id])
+
+        return self.midplane_ip
+
+    def is_midplane_reachable(self):
+        """
+        Retrieves the reachability status of the module from the Supervisor or
+        of the Supervisor from the module via the midplane of the modular chassis
+        Returns:
+            A bool value, should return True if module is reachable via midplane
+        """
+        if not self._is_midplane_up():
+            return False
+
+        command = ['ping', '-c', '1', '-W', '1', self.get_midplane_ip()]
+        try:
+            return subprocess.call(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL) == 0
+        except Exception:
+            return False
+
+    def _is_midplane_up(self):
+        if not self.midplane_interface:
+            platform_dpus_data = DeviceDataManager.get_platform_dpus_data()
+            self.midplane_interface = platform_dpus_data[self.get_name().lower()]["midplane_interface"]
+
+        return utils.read_str_from_file(f'/sys/class/net/{self.midplane_interface}/operstate') == "up"
+
+    def _is_dataplane_online(self):
+        """Check if the dataplane interface is online by querying APPL_DB
+        """
+        if not self.npu_interface:
+            platform_dpus_data = DeviceDataManager.get_platform_dpus_data()
+            npu_dpu_mapping = platform_dpus_data[self.get_name().lower()]["interface"]
+            self.npu_interface = list(npu_dpu_mapping.keys())[0]
+        oper_status = self.app_db.get("APPL_DB", "PORT_TABLE:" + self.npu_interface, "oper_status")
```

Comment:
can you guarantee that redis service will be up earlier than calling this function? And is there other way to check the port oper status instead of accessing redis? From design perspective, we usually avoid calling DB in platform API, platform API should depend on lower layer and static files only.